sycl/include/sycl/ext/oneapi/bindless_images.hpp

```diff
@@ -792,6 +804,43 @@ template <typename DataT> constexpr bool is_recognized_standard_type() {
           std::is_floating_point_v<DataT> || std::is_same_v<DataT, sycl::half>);
 }
 
+#ifdef __SYCL_DEVICE_ONLY__
+
+// Macros are required because it is not legal for a function to return
+// a variable of type 'opencl_image_type'.
+#if defined(__NVPTX__)
+#define CONVERT_HANDLE_TO_IMAGE(raw_handle, ImageType) raw_handle
+#elif defined(__SPIR__)
+#define CONVERT_HANDLE_TO_IMAGE(raw_handle, ImageType)                         \
+  __spirv_ConvertHandleToImageINTEL<ImageType>(raw_handle)
+#else
+#define CONVERT_HANDLE_TO_IMAGE(raw_handle, ImageType) raw_handle
```

Comment:
merge `__NVPTX__` and `else` paths since the code is the same?

Reply:
In my head I was thinking to keep it separated because of different backend like hip, but yeah way to early to think about that. I have merged the paths.